### PR TITLE
Bump toolchain to 1.22

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: .github/workflows/go_test.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.21.x, 1.22.x]
+        go-version: [1.22.x]
```
Comment (roger2hk) [inline .github/workflows/go_test.yml:9]:
```suggestion
        go-version: [1.22.x, 1.23.x]
```

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,8 @@ module github.com/transparency-dev/witness
 
 go 1.21.0
```
Comment (roger2hk) [inline go.mod:3]:
```suggestion
go 1.22.7
```
Bumping the go version here, so we don't need to define the toolchain?

**Review by roger2hk — APPROVED** [review]

**Review by roger2hk — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (AlCutter) [reply go.mod:3]: I think that would mean this module cannot be built with anything below 1.22.7 (as opposed to "you _can_ build with any of the 1.22.0 or higher tools, but we prefer 1.22.6"), which seems quite strict.

**Review by roger2hk — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (roger2hk) [reply go.mod:3]: Sure, that's reasonable.

**Pull request merged**